cli-nuget.js

```diff
@@ -84,6 +88,104 @@ const nugetPublish = {
     },
 }
 
+const nugetUpgrade = {
+    description() {
+        return "Prompts the user to specify a NuGet package to upgrade for all projects in a solution.";
+    },
+    findCsprojFiles() {
+        const solutionDir = dotnetPath.solutionDir();
+
+        echo.message("Looking for csproj files under the current directory...");
+
+        const findResult = shell.find(`${solutionDir}/**/*.csproj`);
+        if (findResult.code !== 0) {
+            echo.error(ERROR_READING_CSPROJ_FILES);
+            shell.exit(findResult.code);
+        }
+
+        return findResult;
+    },
+    getCsprojFilesContainingPackage(csprojFiles) {
+        echo.message(`Looking for packages matching '${this.packageName}'...`);
+
+        const grepResult = shell.grep("-l", this.packageName, csprojFiles);
+        if (grepResult.code !== 0) {
+            echo.error(ERROR_READING_CSPROJ_FILES);
+            shell.exit(grepResult.code);
+        }
+
+        const matchingProjects = grepResult.stdout.split("\n").filter((result) => result.trim() !== "");
+        if (matchingProjects.length === 0) {
+            echo.message(`No projects found with package '${this.packageName}'. Exiting.`);
+            shell.exit(-1);
```

Comment:
Why negative? I believe '1' is the general catch all conventionally

Reply:
Good point 👍 Updated